src/agent/__tests__/MessageSender.test.ts

```diff
@@ -47,50 +47,93 @@ describe('MessageSender', () => {
+  const firstDidCommService = new DidCommService({
+    id: `<did>;indy`,
+    serviceEndpoint: 'https://www.first-endpoint.com',
+    recipientKeys: ['verkey'],
+  })
+  const secondDidCommService = new DidCommService({
+    id: `<did>;indy`,
+    serviceEndpoint: 'https://www.second-endpoint.com',
+    recipientKeys: ['verkey'],
+  })
+  const transportServiceFindServicesMock = mockFunction(transportService.findServices)
 
   let messageSender: MessageSender
   let outboundTransporter: OutboundTransporter
   let connection: ConnectionRecord
+  let outboundMessage: OutboundMessage
 
   describe('sendMessage', () => {
     beforeEach(() => {
       outboundTransporter = new DummyOutboundTransporter()
       messageSender = new MessageSender(enveloperService, transportService, logger)
       connection = getMockConnection({ id: 'test-123' })
+
+      outboundMessage = createOutboundMessage(connection, new AgentMessage())
+
+      envelopeServicePackMessageMock.mockReturnValue(Promise.resolve(wireMessage))
+      transportServiceFindServicesMock.mockReturnValue([firstDidCommService, secondDidCommService])
+      transportServiceFindSessionMock.mockReturnValue(session)
+    })
+
+    afterEach(() => {
+      jest.resetAllMocks()
     })
 
     test('throws error when there is no outbound transport', async () => {
-      const message = new AgentMessage()
-      const outboundMessage = createOutboundMessage(connection, message)
       await expect(messageSender.sendMessage(outboundMessage)).rejects.toThrow(`Agent has no outbound transporter!`)
     })
 
-    test('calls transporter with connection, payload and endpoint', async () => {
-      const message = new AgentMessage()
-      const spy = jest.spyOn(outboundTransporter, 'sendMessage')
-      const outboundMessage = createOutboundMessage(connection, message)
+    test('throws error when there is no service', async () => {
       messageSender.setOutboundTransporter(outboundTransporter)
+      transportServiceFindServicesMock.mockReturnValue([])
+
+      await expect(messageSender.sendMessage(outboundMessage)).rejects.toThrow(
+        `Connection with id test-123 has no service!`
+      )
+    })
+
+    test('calls send message with connection, payload and endpoint from first DidComm service', async () => {
+      messageSender.setOutboundTransporter(outboundTransporter)
+      const sendMessageSpy = jest.spyOn(outboundTransporter, 'sendMessage')
 
       await messageSender.sendMessage(outboundMessage)
 
-      const [[sendMessageCall]] = spy.mock.calls
-      expect(sendMessageCall).toEqual({
+      const [[sendMessageCallArg]] = sendMessageSpy.mock.calls
+      expect(sendMessageCallArg).toEqual({
         connection,
         payload: wireMessage,
-        endpoint,
+        endpoint: firstDidCommService.serviceEndpoint,
         responseRequested: false,
         session,
       })
+      expect(sendMessageSpy.mock.calls.length).toEqual(1)
+    })
```

Comment:
I prefer to use:

`expect(sendMessageSpy).toHaveBeenCalledNthWith()` (not sure if the syntax is 100% correct.

Reply:
Ah. That's right, it's much better. I don't know why I'm not using that 🤷‍♂️